dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserQueryParams.java

```diff
@@ -101,241 +125,91 @@
-            .add( "disabled", disabled ).toString();
-    }
-
     // -------------------------------------------------------------------------
     // Logic
     // -------------------------------------------------------------------------
 
     public UserQueryParams addOrganisationUnit( OrganisationUnit unit )
     {
         this.organisationUnits.add( unit );
+        setOrgUnitBoundary( UserOrgUnitType.DATA_CAPTURE );
         return this;
     }
 
     public UserQueryParams addDataViewOrganisationUnit( OrganisationUnit unit )
     {
-        this.dataViewOrganisationUnits.add( unit );
+
+        this.organisationUnits.add( unit );
+        setOrgUnitBoundary( UserOrgUnitType.DATA_OUTPUT );
         return this;
     }
 
     public UserQueryParams addTeiSearchOrganisationUnit( OrganisationUnit unit )
     {
-        this.teiSearchOrganisationUnits.add( unit );
+        this.organisationUnits.add( unit );
+        setOrgUnitBoundary( UserOrgUnitType.TEI_SEARCH );
         return this;
     }
 
     public boolean hasOrganisationUnits()
     {
-        return !organisationUnits.isEmpty();
+        return !organisationUnits.isEmpty() && orgUnitBoundary == UserOrgUnitType.DATA_CAPTURE;
     }
 
     public boolean hasDataViewOrganisationUnits()
     {
-        return !dataViewOrganisationUnits.isEmpty();
+        return !organisationUnits.isEmpty() && orgUnitBoundary == UserOrgUnitType.DATA_OUTPUT;
     }
 
     public boolean hasTeiSearchOrganisationUnits()
     {
-        return !teiSearchOrganisationUnits.isEmpty();
-    }
-
-    public boolean hasUserGroups()
-    {
-        return !userGroups.isEmpty();
-    }
-
-    public boolean hasUser()
-    {
-        return user != null;
-    }
-
-    // -------------------------------------------------------------------------
-    // Set methods
-    // -------------------------------------------------------------------------
-
-    public UserQueryParams setQuery( String query )
-    {
-        this.query = query;
-        return this;
-    }
-
-    public UserQueryParams setPhoneNumber( String phoneNumber )
-    {
-        this.phoneNumber = phoneNumber;
-        return this;
-    }
-
-    public UserQueryParams setUser( User user )
-    {
-        this.user = user;
-        return this;
-    }
-
-    public UserQueryParams setCanManage( boolean canManage )
-    {
-        this.canManage = canManage;
-        return this;
-    }
-
-    public UserQueryParams setAuthSubset( boolean authSubset )
-    {
-        this.authSubset = authSubset;
-        return this;
+        return !organisationUnits.isEmpty() && orgUnitBoundary == UserOrgUnitType.TEI_SEARCH;
     }
 
-    public UserQueryParams setDisjointRoles( boolean disjointRoles )
+    public UserQueryParams setOrganisationUnits( Set<OrganisationUnit> units )
     {
-        this.disjointRoles = disjointRoles;
+        this.organisationUnits = units;
+        if ( orgUnitBoundary == null )
```

Comment:
Note that this setter is now a bit ambiguous. In the old semantics this was explicitly filtering data capture. In the new semantics this setter is potentially used for any of the boundaries so we just set the type of boundary if it is still null. This is so this can be used in combination with `setOrgUnitBoundary` in any order.